packages/openid4vc-holder/src/presentations/OpenId4VpHolderService.ts

```diff
@@ -0,0 +1,236 @@
+import type { PresentationSubmission } from './selection'
+import type { CredentialsForInputDescriptor } from './selection/types'
+import type { AgentContext, W3cCredentialRecord, W3cVerifiablePresentation } from '@aries-framework/core'
+import type {
+  DIDDocument,
+  PresentationDefinitionWithLocation,
+  SigningAlgo,
+  URI,
+  Verification,
+  VerifiedAuthorizationRequest,
+} from '@sphereon/did-auth-siop'
+import type { PresentationDefinitionV1 } from '@sphereon/pex-models'
+import type { W3CVerifiablePresentation } from '@sphereon/ssi-types'
+
+import {
+  AriesFrameworkError,
+  Buffer,
+  DidsApi,
+  getJwkClassFromKeyType,
+  getKeyFromVerificationMethod,
+  injectable,
+  TypedArrayEncoder,
+  W3cJsonLdVerifiablePresentation,
+  asArray,
+} from '@aries-framework/core'
+import { CheckLinkedDomain, OP, ResponseMode, SupportedVersion, VerificationMode } from '@sphereon/did-auth-siop'
+
+import { PresentationExchangeService } from './PresentationExchangeService'
+
+/**
+ * SIOPv2 Authorization Request with a single v1 presentation definition
+ */
+export type VerifiedAuthorizationRequestWithPresentationDefinition = VerifiedAuthorizationRequest & {
+  presentationDefinitions: [PresentationDefinitionWithLocation & { definition: PresentationDefinitionV1 }]
+}
+
+function isVerifiedAuthorizationRequestWithPresentationDefinition(
+  request: VerifiedAuthorizationRequest
+): request is VerifiedAuthorizationRequestWithPresentationDefinition {
+  return (
+    request.presentationDefinitions !== undefined &&
+    request.presentationDefinitions.length === 1 &&
+    request.presentationDefinitions?.[0]?.definition !== undefined
+  )
+}
+
+@injectable()
+export class OpenId4VpHolderService {
+  public constructor(private presentationExchangeService: PresentationExchangeService) {}
+
+  private getOp(agentContext: AgentContext) {
```

Comment:
Same comment about `OP` being vague.

```suggestion
  private getOpenIdProvider(agentContext: AgentContext) {
```